test/e2e-go/e2e_go_olm_test.go

```diff
@@ -16,36 +16,53 @@ package e2e_go_test
 
 import (
 	"os/exec"
+	"path/filepath"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	"github.com/operator-framework/operator-sdk/internal/testutils"
 )
 
 var _ = Describe("Integrating Go Projects with OLM", func() {
-	Context("with operator-sdk", func() {
-		const operatorVersion = "0.0.1"
 
-		It("should generate and run a valid OLM bundle and packagemanifests", func() {
-			By("turning off interactive prompts for all generation tasks.")
-			err := tc.DisableManifestsInteractiveMode()
-			Expect(err).NotTo(HaveOccurred())
+	// OLM does not work with cert-manager. In this way, we need to generate
+	// the pkg manifest after we comment cert-manager options into the kustomization file.
+	// More info: https://olm.operatorframework.io/docs/advanced-tasks/adding-admission-and-conversion-webhooks/#certificate-authority-requirements
+	BeforeEach(func() {
+		By("commenting cert-manager")
+		err := testutils.ReplaceInFile(
+			filepath.Join(tc.Dir, "config", "default", "kustomization.yaml"),
+			"- ../certmanager", "#- ../certmanager")
+		Expect(err).NotTo(HaveOccurred())
 
-			By("building the bundle")
-			err = tc.Make("bundle", "IMG="+tc.ImageName)
-			Expect(err).NotTo(HaveOccurred())
+		By("commenting cert-manager")
+		err = testutils.ReplaceInFile(
+			filepath.Join(tc.Dir, "config", "default", "kustomization.yaml"),
+			uncommentedCertManager, commentedCertMaagerKustomizeFields)
+		Expect(err).NotTo(HaveOccurred())
+	})
 
-			By("building the operator bundle image")
-			err = tc.Make("bundle-build", "BUNDLE_IMG="+tc.BundleImageName)
-			Expect(err).NotTo(HaveOccurred())
+	AfterEach(func() {
+		By("commenting cert-manager")
+		err := testutils.ReplaceInFile(
+			filepath.Join(tc.Dir, "config", "default", "kustomization.yaml"),
+			"#- ../certmanager", "- ../certmanager")
+		Expect(err).NotTo(HaveOccurred())
 
-			if tc.IsRunningOnKind() {
-				By("loading the bundle image into Kind cluster")
-				err = tc.LoadImageToKindClusterWithName(tc.BundleImageName)
-				Expect(err).NotTo(HaveOccurred())
-			}
+		By("commenting cert-manager")
+		err = testutils.ReplaceInFile(
+			filepath.Join(tc.Dir, "config", "default", "kustomization.yaml"),
+			commentedCertMaagerKustomizeFields, uncommentedCertManager)
+		Expect(err).NotTo(HaveOccurred())
+	})
```

Comment:
Arent we including tests for validating webhook here? We uncomment the cert-manager, run the package manifests and test if everything goes well. But do we also test if the webhooks work? Maybe I am missing something here.

Reply:
The test `e2e_go_olm_test.go` just the integration with OLM the operator tests are in `e2e_go_cluster_test.go`

Reply:
Since webhooks are managed by OLM we should be testing whether they work. Something for a follow-up.